tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh

```diff
@@ -39,9 +41,12 @@ docker build -q -t build-kata-deploy \
 	"${script_dir}/dockerbuild/"
 
 docker run \
+	-v $HOME/.docker:/root/.docker \
 	-v /var/run/docker.sock:/var/run/docker.sock \
 	--env CI="${CI:-}" \
 	--env USER=${USER} -v "${kata_dir}:${kata_dir}" \
 	--rm \
 	-w ${script_dir} \
 	build-kata-deploy "${kata_deploy_create}" $@
+
+rmdir --ignore-fail-on-non-empty "$HOME/.docker"
```

Comment:
`$HOME/.docker` doesn't exist then it is created in line 34. Suppose that on 2nd container some file get created in that directory, thus, this line won't remove  `$HOME/.docker` (but it should). Sorry I am being a little paranoid here, this situation might not happen.... but... you never know for sure :)

Reply:
I'm not aware of any docker command that would create a file in that directory, particularly considering the use-cases we have.

What's your preference?  Add a new flag together with the directory creation and then remove based on the flag?

Reply:
```sh
diff --git a/tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh b/tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh
index 038c917cf..b98434fa5 100755
--- a/tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh
+++ b/tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh
@@ -31,7 +31,11 @@ if [ ${docker_gid} == ${gid} ]; then
        docker_gid=""
 fi
 
-[ -d "$HOME/.docker" ] || mkdir $HOME/.docker
+remove_dot_docker_dir=false
+if [ ! -d "$HOME/.docker" ]; then
+       mkdir $HOME/.docker
+       remove_dot_docker_dir=true
+fi
 
 docker build -q -t build-kata-deploy \
        --build-arg IMG_USER="${USER}" \
@@ -49,4 +53,6 @@ docker run \
        -w ${script_dir} \
        build-kata-deploy "${kata_deploy_create}" $@
 
-rmdir --ignore-fail-on-non-empty "$HOME/.docker"
+if [ $remove_dot_docker_dir ]; then
+       rm -rf "$HOME/.docker"
+fi
```

Reply:
Yes, I thought about using a flag.
`+if [ $remove_dot_docker_dir ]; then` does shell eval this correctly if remove_dot_docker_dir is false/true (instead of 0 or 1)?

Reply:
```
+if [ $remove_dot_docker_dir == true ]; then
+       rm -rf "$HOME/.docker"
+fi
```

This is the correct check :-)

Reply:
Let me update the PR